src/v/cluster_link/task.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "absl/container/flat_hash_map.h"
+#include "base/seastarx.h"
+#include "cluster_link/errc.h"
+#include "cluster_link/model/types.h"
+#include "utils/prefix_logger.h"
+
+#include <seastar/core/condition-variable.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/util/bool_class.hh>
+
+#include <fmt/format.h>
+
+namespace cluster_link {
+
+/**
+ * @brief Abstract class that defines a task in cluster linking
+ */
+class task {
```

Comment:
this is cool, btw. would be nice to have something like this in `util` so that we started building tasks and background fibers using a common framework.